---
title: "Frontpage of presentation"
author: "Mie, Naja, Sille, Sofia"
date: "10-May-2021"
output: ioslides_presentation
---

```{r Install, include=FALSE}
library("tidyverse")
library("usethis")
library("devtools")
library("patchwork")
library(modelr)
options(na.action = na.warn)
library(lubridate)
library(broom)
library(purrr)
library(vroom)
library(stringr)
library(kableExtra)
```

```{r load of model generel to be used in introduction, include=FALSE} 
setwd('../')
source(file = "R/04_model_generel.R", local = knitr::knit_global())
```

# Introduction 
## Data
- The breast cancer data consists of 4 cancer groups, and 3 healthy patients as well as 
```{r show cancer class and other relevant, echo=FALSE}
joined_data %>%
  group_by(`Class`, Gender) %>%
  count()
```
other relevant parameters. (We will get into these in method)

## Goals
- Exploration and identifying patterns in the data
- Create visualization of the data 
- Predict breast cancer class by looking at specific gene expressions

# Materials and methods
## Materials
### Data
- 77_cancer_proteomes_CPTAC_itraq.csv
  - Reference Sequence protein ID
  - Gene symbol
  - Gene name

- Clinical_data_breast_cancer.csv
  - Patient ID (TCGA ID)
  - Gender
  - Cancer class
  - Other relevant clinical observation

[Source: Kaggle](https://www.kaggle.com/piotrgrabo/breastcancerproteomes#clinical_data_breast_cancer.csv)  
  
## Materials
### Extract of Raw data
```{r load raw data, include=FALSE}
setwd('../')
source(file = "R/01_load.R", knitr::knit_global())
```

```{r show raw data, echo=FALSE}
set.seed(77777) #added to ensure same slice_sample choose the same 4 rows each time.
proteomes %>%
  select(1:4) %>%
  slice_sample(n =4) %>%
  kable("html") %>%
  kable_styling(font_size = 10)

clinical %>%
  select(`Complete TCGA ID`, Gender, `Age at Initial Pathologic Diagnosis`, Tumor, `PAM50 mRNA`) %>% #add flere columns if relevant
  slice(1:4) %>%
  kable("html") %>%
  kable_styling(font_size = 10)
```

## Methods: Clearing of data
### We have done some clearning of the two data sets 
- Proteomes: 
  - Rename, so the column names aligns with the clinical file
  - Removing .01TCGA from every column name in to match ID's in clinical using rename() in combination with str_replace_all()
  - Removing duplicates using select() 
  
- Clinical: 
  - Aligning ID's with proteome file using sub_str()


## Methods: Wranged data
### After the clearning, the datasets were wranged and joined
- Proteomes: 
  - Removing fraction NA > 0.15 using filter()
  - Replace the remaining NAs with median of column
  - Transpose data (get protein ID as columns)

- Clinical
  - Rename of PAM50 mRNA to Class

Finally, a right_join() with x = clinical, y = proteomes, and by patient ID (TCGA_ID) was used to make the joined_data. For the joined data, the age was divided into agegroups using mutate and casewhen(). A joined_healthy_data consists on only the heathly was also made.

## Methods: Packages
- tidyverse
  - dplyr
  - ggplot2
  - broom
  - stringr
  - tidymodels

- others
  - patchwork
  - purrr
  - vroom
  - kableExtra

# Results
## Removal for genes according to the fraction of NAs
```{r load and plot FracNA, echo=FALSE, warning=FALSE, message=FALSE} 
setwd('../')
source(file = "R/04_model_generel.R", knitr::knit_global())

fractionNA_All + fractionNA_without0
```

## Removal for genes according to the fraction of NAs comments
Lots of the represented genes expression do not contain NA vlaues. Thereby, all of them was removed from the plot, to get a better estimate of where to cut the lowest wanted fraction of NAs.For this, we end up removing 2205 genes.
```{r removed FracNA, echo=FALSE, warning=FALSE, message=FALSE} 
proteomes_clean %>%
  filter(Frac_NA > 0.25) %>%
  count()
```

## Heart map
```{r load file of Heatmap, include=FALSE} 
setwd('../')
source(file = "R/04_model_HM.R", local = knitr::knit_global())
```

Two Heat maps are plotted using geom_tile() on relevant Breasts Cancer Genes. One is based on tumor size whereas the other is based on class
```{r show HeatMap, echo=FALSE, warning=FALSE, message=FALSE}
HM_class + HM_TumorSize
```


## PCA 


## Discussion
- The removed duplicates were selected from an error when aligning ID's. This was done by se R filen 02 hvis relevant; HVIS NU DET ER RELEVANT AT HAVE MED.
- Fraction of NA > 0.15 has been removed based on the plot showed in results. 
  - Calculated by rowSums(is.na(select(.,-RefSeqProteinID)))/80)
- Discuss the selection of cancer_genes.
